readme.md

```diff
@@ -0,0 +1,28 @@
+### Hetmec Front-end
+
+This repository contains the front-end interface for the [Hetmech project](https://github.com/greenelab/hetmech).
+
+
+#### Test/Build
+
+To install environment:
+
+1. Install latest version of the [Node.js platform](https://nodejs.org/en/)
+2. Clone this repo
+3. Navigate to your local clone
+4. Run `npm install` to download and install all the node package dependencies
+
+To build packaged/bundled app:
+
+1. Run `npx webpack`
```

Comment:
I am not sure whether it is a good idea to convert the building process from React's `react-scripts` to the native webpack. (The former actually calls the latter, but hides all webpack config details.)

By the way, I ran `npx webpack` command on my desktop and the building seems okay but the terminal prompt never comes back. Is it supposed to happen?

Reply:
You may also want to rename this file into `README.md`, which is more widely used.

Reply:
`react-scripts` calls many things we won't need in our project. It has close to 50 dependencies, most of which are listed as in-package (as opposed to dev) dependencies. So that would dramatically increase the size of the built package with a lot of stuff we wont use.

As a side note, I'm not sure why npm.com is showing the "*-loader" dependencies as non-dev dependencies. I wonder if that is a mistake, because to my knowledge, those types of things are only relevant in the build step.

Reply:
Posting private convo with @dongbohu here for posterity. Convo is in regards to the build size of a "from-scratch" webpack config vs `react-scripts`.

> so the problem is, almost all of the dependencies listed in react-scripts are being used and are necessary for the build because of the way they've set up the example project and config

> i could only remove about 5 of them without breaking the build
> but for the 5 i did remove, the size of the build didn't change significantly
> here is the size comparison between the from scratch and react-scripts though:

> *vanilla/from scratch*
> html 163 bytes
> js 119 kb
> ./build 119 kb

> *nonejected, ejected w/ ~5 dependencies removed*
> html 2 kb
> js total 481 kb
> ./build 513 kb (contribtion of images to size is negligible)